src/wp-includes/media.php

```diff
@@ -5011,9 +5012,128 @@ function wp_getimagesize( $filename, array &$image_info = null ) {
 	 */
 	if ( 2 === func_num_args() ) {
 		// phpcs:ignore WordPress.PHP.NoSilencedErrors
-		return @getimagesize( $filename, $image_info );
+		return @_wp_get_image_size( $filename, $image_info );
 	} else {
 		// phpcs:ignore WordPress.PHP.NoSilencedErrors
-		return @getimagesize( $filename );
+		return @_wp_get_image_size( $filename );
 	}
 }
+
+/**
+ * Extract meta information about a webp file: width, height and type.
+ *
+ * @param [type] $filename Path to a WebP file.
+ * @return array $webp_info {
+ *     An array of WebP image information.
+ *
+ *     @type array $size {
+ *         @type int  $width  Image width.
+ *         @type int  $height Image height.
+ *         @type bool $type   The WebP type: one of 'lossy', 'lossless' or 'animated-alpha'.
+ *     }
+
+ */
+function wp_get_webp_info( $filename ) {
+	$width  = false;
+	$height = false;
+	$type   = false;
+	if ( ! 'image/webp' === wp_get_image_mime( $filename ) ) {
+		return compact( 'width', 'height', 'type' );
+	}
+	try {
+		$handle = fopen( $filename, 'rb' );
+		if ( $handle ) {
+			$magic = fread( $handle, 40 );
+			fclose( $handle );
+
+			// Make sure we got enough bytes.
+			if ( strlen( $magic ) < 40 ) {
+				return compact( 'width', 'height', 'type' );
+			}
+
+			// The headers are a little different for each of the three formats.
+			switch ( substr( $magic, 12, 4 ) ) {
+				// Lossy WebP.
+				case 'VP8 ':
+					$parts  = unpack( 'v2', substr( $magic, 26, 4 ) );
+					$width  = (int) ( $parts[1] & 0x3FFF );
+					$height = (int) ( $parts[2] & 0x3FFF );
+					$type   = 'lossy';
+					break;
+				// Lossless WebP.
+				case 'VP8L':
+					$parts  = unpack( 'C4', substr( $magic, 21, 4 ) );
+					$width  = (int) ( $parts[1] | ( ( $parts[2] & 0x3F ) << 8 ) ) + 1;
+					$height = (int) ( ( ( $parts[2] & 0xC0 ) >> 6 ) | ( $parts[3] << 2 ) | ( ( $parts[4] & 0x03 ) << 10 ) ) + 1;
+					$type   = 'lossless';
+					break;
+				// Animated/alpha WebP.
+				case 'VP8X':
+					// Pad 24-bit int.
+					$width = unpack( 'V', substr( $magic, 24, 3 ) . "\x00" );
+					$width = (int) ( $width[1] & 0xFFFFFF ) + 1;
+					// Pad 24-bit int.
+					$height = unpack( 'V', substr( $magic, 27, 3 ) . "\x00" );
+					$height = (int) ( $height[1] & 0xFFFFFF ) + 1;
+					$type   = 'animated-alpha';
+					break;
+			}
+		}
+	} catch ( Exception $e ) {
+	}
+	return compact( 'width', 'height', 'type' );
+}
+
+/**
+ * Determine if a passed image is a lossy WebP image.
```

Comment:
While not all of core does this, documentation standards state that function docs should start with a third-person verb.
```suggestion
 * Determines if a passed image is a lossy WebP image.
```